apps/prairielearn/src/pages/instructorStudents/components/ColumnManager.tsx

```diff
@@ -0,0 +1,214 @@
+
+    // eslint-disable-next-line react-you-might-not-need-an-effect/no-event-handler
+    if (activeElementId) {
+      const element = document.getElementById(activeElementId);
+      // eslint-disable-next-line react-you-might-not-need-an-effect/no-event-handler
+      if (element) {
+        (element as HTMLElement).focus();
+      }
+    }
+  }, [activeElementId]);
+
+  return (
+    <Dropdown
+      ref={menuRef}
+      autoClose="outside"
+      show={dropdownOpen}
+      onToggle={(isOpen, _meta) => setDropdownOpen(isOpen)}
+      onFocusOut={(e: FocusEvent) => {
+        // Since we aren't using role="menu", we need to manually close the dropdown when focus leaves.
+        if (menuRef.current && !menuRef.current.contains(e.target as Node)) {
+          setDropdownOpen(false);
+        }
+      }}
+    >
+      <Dropdown.Toggle variant="outline-secondary" id="column-manager-button">
+        <i class="bi bi-view-list me-2" aria-hidden="true" />
+        View
+      </Dropdown.Toggle>
+      <Dropdown.Menu>
+        {pinnedColumns.length > 0 && (
+          <>
+            <div class="px-2 py-1 text-muted small" role="presentation">
+              Frozen columns
+            </div>
+            <div role="group">
+              {pinnedColumns.map((column, index) => {
+                return (
+                  <ColumnMenuItem
+                    key={column.id}
+                    column={column}
+                    hidePinButton={index !== pinnedColumns.length - 1}
+                    onTogglePin={handleTogglePin}
+                    onClearElementFocus={() => setActiveElementId(null)}
+                  />
+                );
+              })}
+            </div>
+            <Dropdown.Divider />
+          </>
+        )}
+        {unpinnedColumns.length > 0 && (
+          <>
+            <div role="group">
+              {unpinnedColumns.map((column, index) => {
+                return (
+                  <ColumnMenuItem
+                    key={column.id}
+                    column={column}
+                    hidePinButton={index !== 0}
+                    onTogglePin={handleTogglePin}
+                    onClearElementFocus={() => setActiveElementId(null)}
+                  />
+                );
+              })}
+            </div>
+            {showResetButton && <Dropdown.Divider />}
+          </>
+        )}
+        {showResetButton && (
+          <div class="px-2 py-1">
+            <Button
+              variant="secondary"
+              size="sm"
+              class="w-100"
+              aria-label="Reset all columns to default visibility and pinning"
+              onClick={() => {
+                table.resetColumnVisibility();
+                table.resetColumnPinning();
+                setActiveElementId('column-manager-button');
+              }}
+            >
+              <i class="bi bi-arrow-counterclockwise me-2" aria-hidden="true" />
+              Reset view
```

Comment:
Nit: I would have expected this button to reset to the *default* state, e.g. "UID" being pinned. Put differently: if I loaded this page and then clicked this button, I would have expected nothing to happen. Instead, it resets to "nothing pinned". Not sure if this is intentional, but it feels kinda weird to me.